DatadogInternal/Tests/MessageBus/FeatureBaggageTests.swift

```diff
@@ -5,97 +5,83 @@
  */
 
 import XCTest
+import TestUtilities
 @testable import DatadogInternal
 
 class FeatureBaggageTests: XCTestCase {
-    private enum EnumAttribute: String, Codable {
-        case test
+    struct GroceryProduct: Encodable, RandomMockable {
+        var name: String
+        var points: Int
+        var description: String?
+
+        static func mockRandom() -> Self {
+            .init(
+                name: .mockRandom(),
+                points: .mockRandom(),
+                description: .mockRandom()
+            )
+        }
     }
 
-    struct SomeEncodable: Encodable {
-        var string: String
+    struct CartItem: Decodable {
+        var name: String
+        var points: Int
     }
 
-    struct SomeDecodable: Decodable {
-        var string: String
-    }
-
-    let attributes: FeatureBaggage = [
-        "int": 1,
-        "string": "test",
-        "double": 2.0,
-        "enum": EnumAttribute.test
-    ]
-
-    func testSubscript() {
-        var attributes = attributes
-        XCTAssertEqual(attributes["int", type: Int.self], 1)
-        XCTAssertEqual(attributes["double", type: Double.self], 2.0)
-        XCTAssertNil(attributes["string", type: Int.self])
-        XCTAssertEqual(attributes["string"], "test")
+    func testEncodeDecode() throws {
+        let pear = GroceryProduct.mockRandom()
+        let baggage = FeatureBaggage(pear)
+        let item: CartItem = try baggage.decode()
 
-        attributes["int"] = 2
-        XCTAssertEqual(attributes["int"], 2)
+        XCTAssertEqual(pear.name, item.name)
+        XCTAssertEqual(pear.points, item.points)
```

Comment:
Worth testing the other way round and seeing if description is empty

Reply:
`item` does not have a `description` property, but I can test an optional property in `CartItem` indeed 👍